src/s11/handlers/s11_msg_delegator.c

```diff
@@ -175,23 +177,23 @@ handle_s11_message(void *message)
 
 	switch(msgHeader.msgType){
 	case S11_GTP_CREATE_SESSION_RESP:
-		s11_CS_resp_handler(msgBuf_p, &msgHeader);
+		s11_CS_resp_handler(msgBuf_p, &msgHeader, sgw_ip);
```

Comment:
see..it has cluttered all across the places... if it remains within message it looks good...

Reply:
Then wouldn't we have to manipulate the read pointer every time we access the buffer?
Another option is to add the ip address after we add the bytes..then it can be decoded in each handler after the entire message has been read.

Reply:
Yeah..decoding in each handler. Thing is as soon as I add support for transactions in s11 container, i will not need sgw_ip in the response messages. It's only required in new request messages initiated by MME. Response always goes to the source who initiated message.